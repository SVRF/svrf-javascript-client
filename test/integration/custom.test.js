import Svrf from '../../src';
import HttpClient from '../../src/http/http-client';

const {SVRF_TEST_API_KEY} = process.env;

describe('custom request', () => {
  describe('svrf.request', () => {
    it('sends custom api request', async () => {
      const api = new Svrf(SVRF_TEST_API_KEY);
      const result = await api.request('get', 'vr/trending');

      expect(result.success).toEqual(true);
      expect(result.media.length > 0).toEqual(true);
    });

    it('uses provided http client', async () => {
      const api = new Svrf('', {isManualAuthentication: true});
      const httpClient = new HttpClient();

      await expect(api.request('get', 'vr/trending', {httpClient})).rejects.toThrowErrorMatchingSnapshot();
    });
  });
});
