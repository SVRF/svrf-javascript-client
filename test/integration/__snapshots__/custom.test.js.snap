// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom request svrf.request uses provided http client 1`] = `"Request failed with status code 401"`;
